controllers/user.js

```diff
@@ -32,30 +33,23 @@ exports.signup = function(req, res, next) {
         if (!isEmpty(errors_ret)) {
             rerender_signup(errors_ret, req, res, next);
         } else {
-            return models.Users.findOne({
-                where: {
-                    is_admin: true
-                }
-            }).then(user => {
+            return new Promise(function(resolve, reject) {
                 let newUser;
                 newUser = models.Users.build({
                     email: req.body.email,
                     password: generateHash(req.body.password),
                     is_admin: false
                 });
                 return newUser.save().then(result => {
-                    passport.authenticate('local', {
-                        successRedirect: "/",
-                        failureRedirect: "/signup",
-                        failureFlash: true
-                    })(req, res, next);
+                    exports.login(req, res, next);
```

Comment:
Checking the purpose of this line?
login will call passport.authenticate again

Reply:
yes, I have wrapped the authentication in a separate function, since the same code was repeated more the once. Now the function is simply called when we need to authenticate

Reply:
Actually, I found a problem in this solution: the two calls for authentication are not the same. Here in case of failure it will redirect to the signup page. I will revert this